DOCSP-37494 Get started (#38)
mongodb/docs-pymongo: source/get-started/download-and-install.txt

.. _pymongo-get-started-download-and-install:

====================
Download and Install
====================

.. procedure::
   :style: connected

   .. step:: Install Dependencies

      Ensure you have Python3 version 3.7 or later,
      pip, and dnspython installed in your development environment.

>       For information on how to install pip, see the

Reviewer: ```suggestion
      For information about how to install pip, see the
```

      `pip Installation guide <https://pip.pypa.io/en/stable/installation/>`__.

      Support for ``mongodb+srv://`` connection URIs requires dnspython.

>       For information on how to install dnspython, see the `dnspython

Reviewer: ```suggestion
      For information about how to install dnspython, see the `dnspython
```

>       Ensure you have Python3 version 3.7 or later,
>       pip, and dnspython installed in your development environment.
> 
>       For information about how to install pip, see the
>       `pip Installation guide <https://pip.pypa.io/en/stable/installation/>`__.
> 
>       Support for ``mongodb+srv://`` connection URIs requires dnspython.
>       For information about how to install dnspython, see the `dnspython
>       PyPi project page <https://pypi.org/project/dnspython/>`__.

Reviewer: S: What if this were a list with links? Thinking it might make it clearer + stick to the style guide recs for links

```suggestion
      Ensure you have the following dependencies installed
      in your development environment:
      
       - **Python3 version 3.7 or later.** To learn more about installing Python, see
         `Download Python. <https://www.python.org/downloads/>`__
       - **pip.** For information about how to install pip, see
         `Installation <https://pip.pypa.io/en/stable/installation/>`__ in the pip
         documentation.
         - **dnspython.** Support for ``mongodb+srv://`` connection URIs requires dnspython.
            For information about how to install dnspython, see the `dnspython 
            <https://pypi.org/project/dnspython/>`__ PyPi page.
```

Author: Good call!

   .. step:: Create a Project Directory

      In your shell, run the following command to create a 
      directory called ``pymongo-quickstart`` for this project:

      .. code-block:: bash

         mkdir pymongo-quickstart

      Run the following command to navigate into the project 
      directory:

      .. code-block:: bash

>          cd pymongo_quickstart

Reviewer: Previous line of code uses a hyphen, not an underscore. not sure which is more idiomatic for python directories

```suggestion
         cd pymongo-quickstart
```

      Run the following command to create a file to contain your application:

      .. code-block:: bash

         touch quickstart.py

   .. step:: Install {+driver-short+}

      Run the following command in your shell to install
      the driver in your project directory:

      .. code-block:: bash

         $ python3 -m pip install pymongo=={+patch-version-number+}

> After you complete these steps, you have Python3, pip, dnspython, 
> and a new project directory with the driver dependencies installed.

Reviewer: S:

```suggestion
After you complete these steps, you have a new project directory
and the driver dependencies installed.
```

   .. step:: Install {+driver-short+}

      Run the following commands in your shell to create and activate a virtual
      environment in which to install the driver:

      .. code-block:: bash

>          python -m venv venv

Reviewer 2: Strangely enough, one often doesn't have `python` on their machine, and must start with `python3`. Once a virtual env is activated `python` will always be there (typically symbolically linked to python3). So, the first line should be `python3 -m venv venv`. The rest is good, although, as described, one typically uses plain python in the activated environment. The necessary change is only the one python -> python3.

Author: Good catch, thanks! Fixed
